### DISCUSSION TITLE
SyntaxError: Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by cfcmdev26** [mercadopago/sdk-react#52] [Q&A] [answered]
i am getting this error when trying to implement checkout pro in react specifically in nextjs. 
i have tried to fix it but i have not succeeded. could someone help me?

error : \node_modules\@mercadopago\sdk-react\index.js:1
import initMercadoPago from './mercadoPago/initMercadoPago';

**Comment by francineemilia** [comment]
Hi @mrNotSenior

Could you send us a sample from your code to help us understand what could be causing the error?

**Reply by joaogabriels** [reply]
Hi @francineemilia,

I am currently facing a similar issue with NextJs version 13. I'm encountering an error when attempting to import methods from `@mercadopago/sdk-react` or `@mercadopago/sdk-react/coreMethods`. The error message I'm receiving is the same as what @mrNotSenior  mentioned:

```
index.js:619 Uncaught SyntaxError: Cannot use import statement outside a module
    at /usr/local/src/node_modules/ (mercadopago/sdk-react/coreMethods/index.js:1)
```

I have attempted to reinstall the library and change the way I import the methods, but unfortunately, none of these attempts have been successful so far.

I apologize for not being able to share my code, but I'm more than willing to provide additional information if needed.

**Reply by francineemilia** [reply]
@joaogabriels could you specify which components from coreMethods you're trying to use? And/Or share with us only the parts from your code with the components? I'll try to replicate the error here, but until now I didn't have the same response as yours.

**Reply by joaogabriels** [reply]
Thank you for the quick response!

Currently, my project utilizes only a few core methods from the SDK:

From @mercadopago/sdk-react:

- [initMercadoPago](https://github.com/mercadopago/sdk-react#initialization)

From @mercadopago/sdk-react/coreMethods:

- [getPaymentMethods](https://github.com/mercadopago/sdk-react#getpaymentmethods)
- [createCardToken](https://github.com/mercadopago/sdk-react#createcardtoken-1)

I attempted to import these methods into my ES6 TypeScript class and React components, but unfortunately, they did not work. I followed the documentation and double-checked my installation, and the only difference from the documentation is that I am not using any of the [CheckoutBricks](https://github.com/mercadopago/sdk-react#checkout-bricks) in my application.

I have created a sandbox with some code that reproduces the same error. The only difference from my actual application code is that I imported initMercadoPago inside one of my ES6 TypeScript files, but the error persists.

Here is the link to the sandbox: [Sandbox link](https://codesandbox.io/p/sandbox/peaceful-tereshkova-dtzmxt?welcome=true)

**Reply by francineemilia** [reply]
@joaogabriels I'm still analyzing your case, but I have I doubt: this is a server or a client component? Because if its a server component you cannot use useEffect, but if its a client component you need to use the [Client Component Directive](https://nextjs.org/docs/getting-started/react-essentials#client-components).

When I tried use getPaymentMethod in a client component I did like this:
``` javascript
"use client";
import {
  initMercadoPago,
  getPaymentMethods,
} from "@mercadopago/sdk-react";

async function test() {
  const paymentMethods = await getPaymentMethods({ bin: "503175" });
}
initMercadoPago("<YOUR_PUBLIC_KEY>");

export default function Home() {
  test();
  return <h1>teste</h1>;
}
```

**Reply by joaogabriels** [reply]
@francineemilia, it's a client component. I am using useEffect throughout my application without any issues. The only significant server feature that my application utilizes is Static Site Generation with getStaticProps. However, I ensure that the components are only rendered when the user is accessing a non-fallback page.

This issue seems to be related to an internal problem or an incompatibility with Next.js. Interestingly, when I attempted to use the [js](https://github.com/mercadopago/sdk-js) SDK, it worked perfectly fine.

**Reply by AdaltonLeite** [reply, marked as answer]
Hi @joaogabriels, 

I thing that to render the Mercado Pago - React SDK we need to import it with [Dynamic Import from NextJs](https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#nextdynamic), this enable the code to be ignored by the Server Side and only run on the Client Side. 

Here is an example [Wiki](https://github.com/mercadopago/sdk-react/wiki/Using-Bricks-with-Next.js) of a  where it is possible to render the React SDK in NextJS, using the method `getPaymentMethods`

**Reply by bubamacko** [reply]
> Hi @joaogabriels,
> 
> I thing that to render the Mercado Pago - React SDK we need to import it with [Dynamic Import from NextJs](https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#nextdynamic), this enable the code to be ignored by the Server Side and only run on the Client Side.
> 
> Here is an example of a [codeSandbox ](https://codesandbox.io/p/sandbox/checkout-brick-nextjs-vyz4nq?file=%2Fnext.config.js&selection=%5B%7B%22endColumn%22%3A3%2C%22endLineNumber%22%3A4%2C%22startColumn%22%3A3%2C%22startLineNumber%22%3A4%7D%5D)where it is possible to render the React SDK in NextJS, using the method `getPaymentMethods`

Can you reupload the sandbox¿?

**Reply by AdaltonLeite** [reply]
Hi @bubamacko, 

Unfortunately codesanbox removed this link and we lost the code. We have documentation on the [wiki](https://github.com/mercadopago/sdk-react/wiki/Using-Bricks-with-Next.js) that shows you step by step how to adjust this way of importing the react module into nextjs. I think it might help you.

**Reply by bubamacko** [reply]
Ohh, well, thanks! i'll check it out! Can i ask you something else?  Is there anywhere where i can check where i can get the card token using bricks?

**Comment by longlho** [comment]
I'm getting this issue as well primarily because Jest can't handle ESM import, forcing us to transpile this SDK specifically. Is there a way u guys can distribute a cjs bundle as wel?

**Reply by ignaciolarranaga** [reply]
+1
